translib/Makefile

```diff
@@ -53,6 +54,9 @@ $(TRANSL_DB_TEST_BIN) : $(TRANSL_DB_ALL_SRCS)
 $(TRANSFORMER_TEST_BIN): $(TRANSFORMER_TEST_SRCS) $(TRANSFORMER_SRCS) $(TRANSLIB_MAIN_SRCS) $(YGOT_BINDS)
 	$(GO) test -mod=vendor -tags xfmrtest -c -vet=off -cover -coverpkg=../translib/transformer ../translib/transformer -o $@
 
+$(TRANSFORMER_TEST_APP_BIN): $(TRANSFORMER_TEST_SRCS) $(TRANSFORMER_SRCS) $(TRANSLIB_MAIN_SRCS) $(YGOT_BINDS)
+	$(GO) test -mod=vendor -tags testapp -c -vet=off -cover -coverpkg=../translib/transformer ../translib/transformer -o $@
```

Comment:
@amrutasali I have added this based on our discussion. Could you please provide your inputs.

Reply:
transformer.test  is just  for transformer infrastructure test only. We didn't want to mix app specific code in transformer test to avoid app specific failures to be attributed to infra.

Reply:
We have added the test app to pipelines in azure-pipelines.yml and removed the test app build tags

Reply:
Files without build tags will be part of both test binaries. You cannot achieve the infra & app separation as @amrutasali indicated. Test cases will be run twice as part of both binaries (eg. search "=== RUN   Test_node_sonic_sflow_interface/Enable_sFlow" in the [PR sanity](https://dev.azure.com/mssonic/build/_build/results?buildId=285126&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=c10d5f44-55ce-55d7-7975-407ed75d9a96&l=7519) logs). The 2nd test binary is redundant.

We have got two options:

1) Do not use the 2nd build tag for apps and keep only one test binary. The xfmrtest tag will be used to only to enable extra transformer apps when test yang models are included.

2) Use 2 build tags and 2 test binaries (the original approach used in this PR). Only problem here is that it assumes developers will add correct build tag in every *_test.go files. I feel it will be difficult to enforce this in the community. If someone does not add the tag, the test cases will be part of both tests and we will lose infra & app separation. All PR reviewers and approvers need to watch it.

Reply:
Another option is to use the negation of "xfmrtest" tag to separate tests.
i.e. in the sflow_*_test.go, add the build tag
`// +build !xfmrtest`

In case a community user wants to implement new test apps, one is likely to check the build tag from existing test code as a reference.

Reply:
As we discussed, I have proceeded with the second option of original approach used in the PR. ie using testapp build tag for the application test and added in the PR sanity pipelines.